pkg/ccl/crosscluster/logical/dead_letter_queue.go

```diff
@@ -129,19 +129,35 @@ func (dlq *loggingDeadLetterQueueClient) Log(
 }
 
 type deadLetterQueueClient struct {
-	ie isql.Executor
+	ie              isql.Executor
+	tableIDToPrefix map[int32]string
 }
 
-func (dlq *deadLetterQueueClient) Create(ctx context.Context, tableIDs []int32) error {
-	if _, err := dlq.ie.Exec(ctx, "create-enum", nil, writeEnumStmt); err != nil {
-		return errors.Wrap(err, "failed to create crdb_replication_mutation_type enum")
-	}
+func (dlq *deadLetterQueueClient) Create(ctx context.Context) error {
+	// Store unique db names to avoid executing identical enum statements.
+	uniqueDbNames := make(map[string]bool)
 
-	// Create a dlq table for each given table.
-	for _, tableID := range tableIDs {
-		tableName := fmt.Sprintf(dlqBaseTableName, tableID)
-		if _, err := dlq.ie.Exec(ctx, "create-dlq-table", nil, fmt.Sprintf(createTableStmt, tableName)); err != nil {
-			return errors.Wrapf(err, "failed to create dlq table %q", tableName)
+	// Create a dlq table for each table to be replicated.
+	for tableID, prefix := range dlq.tableIDToPrefix {
+		tableName := fmt.Sprintf(dlqBaseTableName, prefix, tableID)
```

Comment:
nit: i don't think it is necessary to include the tableID in the dlq table name

Reply:
Seems like below we are making a DLQ per table? But then we also are putting the table ID into the DLQ.